src/main/java/edu/harvard/iq/dataverse/api/CustomizationApi.java

```diff
@@ -0,0 +1,29 @@
+package edu.harvard.iq.dataverse.api;
+
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import jakarta.ejb.EJB;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.Response;
+
+@Path("customization")
+public class CustomizationApi extends AbstractApiBean {
+
+    @EJB
+    SettingsServiceBean settingsService;
+
+    @GET
+    @Path("analytics")
+    @Produces({"text/html; charset=UTF-8"})
+    public Response getAnalyticsHTML() {
+        return getFromSettings(SettingsServiceBean.Key.WebAnalyticsCode, "text/html; charset=UTF-8");
+    }
+
+    private Response getFromSettings(SettingsServiceBean.Key key, String type) {
+        String value = settingsService.getValueForKey(key);
+        if (value != null) {
+            return Response.ok(value).type(type).build();
```

Comment:
Since the key is just the URL of the analytics file, should this be a redirect to it? (The key value itself is not html.)

Reply:
Right, or you could call in the path on disk. Like this, from the guides:

`curl -X PUT -d '/var/www/dataverse/branding/analytics-code.html' http://localhost:8080/api/admin/settings/:WebAnalyticsCode`

The idea here is that the HTML is in /var/www/dataverse/branding/analytics-code.html

Like Jim said, all that HTML isn't in the http://localhost:8080/api/admin/settings/:WebAnalyticsCode database setting. It's just a path to the HTML.

Reply:
Good catch - this isn't a relative URL that you could redirect to, it's a file path. However, we already have a URL to retrieve it from: /CustomizationFilesServlet?customFileType=analytics .